presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConfigAPI.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller.api;
+
+import com.google.gson.Gson;
+import com.sun.org.apache.regexp.internal.RE;
+import com.teradata.prestomanager.common.ExtendedStatus;
+import com.teradata.prestomanager.common.NodeSet;
+import com.teradata.prestomanager.controller.ApiRequester;
+import com.teradata.prestomanager.controller.ApiScope;
+import io.swagger.annotations.ApiParam;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.eclipse.jetty.http.HttpMethod;
+
+import javax.inject.Singleton;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Path("/config")
+@Singleton
+public class ControllerConfigAPI
+{
+    private static final Logger LOGGER = LogManager.getLogger(ControllerConfigAPI.class);
+    private NodeSet nodeSet = NodeSet.getInstance();
+
+    @GET
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response getConfig(@QueryParam("scope") @DefaultValue("INVALID") String scope,
+            @QueryParam("nodeId") @DefaultValue("null") String nodeId)
+    {
+        ApiScope apiScope = ApiScope.fromString(scope);
+        try{
+            checkArguments(apiScope, nodeId);
+        }
+        catch (IllegalArgumentException e) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Invalid scope").build();
+        }
+
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("getConfig")
+                .httpMethod(HttpMethod.GET)
```

Comment:
Oh, for some reason I thought this comment was a few lines higher up. That explains my confusion.